test/goth_test.exs

```diff
@@ -26,6 +26,35 @@ defmodule GothTest do
     {:ok, ^token} = Goth.fetch(test)
   end
 
+  test "async fetch", %{test: test} do
+    now = System.system_time(:second)
+    bypass = Bypass.open()
+
+    Bypass.expect(bypass, fn conn ->
+      body = ~s|{"access_token":"dummy","expires_in":3599,"token_type":"Bearer"}|
+      Plug.Conn.resp(conn, 200, body)
+    end)
+
+    config = [
+      name: test,
+      source: {:service_account, random_service_account_credentials(), url: "http://localhost:#{bypass.port}"},
+      async: true
+    ]
+
+    start_supervised!({Goth, config})
+
+    # higher timeouts since calculating JWT is expensive
+    Process.sleep(1000)
```

Comment:
I still doesn't know how to improve this check, tried with `pong` and still fails with `Goth.fetch/1` below

Reply:
handle_continue is always executed after the server already started so yeah, start_supervised cannot wait.